src/python/py_utils.cpp

```diff
@@ -220,6 +220,27 @@ ov::genai::StreamerVariant pystreamer_to_streamer(const PyBindStreamerVariant& p
     return streamer;
 }
 
+ov::genai::ChunkStreamerVariant pystreamer_to_chunk_streamer(const PyBindChunkStreamerVariant& py_streamer) {
+    ov::genai::ChunkStreamerVariant streamer = std::monostate();
+
+    std::visit(overloaded {
+    [&streamer](const std::function<bool(py::str)>& py_callback){
+        // Wrap python streamer with manual utf-8 decoding. Do not rely
+        // on pybind automatic decoding since it raises exceptions on incomplete strings.
+        auto callback_wrapped = [py_callback](std::string subword) -> bool {
+            auto py_str = PyUnicode_DecodeUTF8(subword.data(), subword.length(), "replace");
+            return py_callback(py::reinterpret_borrow<py::str>(py_str));
+        };
+        streamer = callback_wrapped;
+    },
+    [&streamer](std::shared_ptr<ChunkStreamerBase> streamer_cls){
+        streamer = streamer_cls;
+    },
+    [](std::monostate none){ /*streamer is already a monostate */ }
+    }, py_streamer);
```

Comment:
`std::visit` returns whatever lambdas return. You can probably simplify this with `return std::visit(...);` and every lambda should return an instance of `ov::genai::ChunkStreamerVariant`

Reply:
Done:
https://github.com/openvinotoolkit/openvino.genai/pull/1148/files#diff-748b4bf50147f3115ec7d8fc33f852a284e650520248965a532a41c3c514579aR222